core/src/BibleEngine.class.ts

```diff
@@ -717,11 +718,13 @@ export class BibleEngine {
             currentPhraseNum: number;
             sectionLevel: number;
             recursionLevel: number;
+            alreadyUsedPhraseIds: any;
         }
     ): Promise<{ firstPhraseId: number | undefined; lastPhraseId: number | undefined }> {
         const entityManager = await this.pEntityManager;
         let firstPhraseId: number | undefined, lastPhraseId: number | undefined;
-        for (const content of contents) {
+        const validContents = contents.filter(content => content);
```

Comment:
don't know how to feel about this. this checks for something that can't happen when types are respected everywhere. in the end that's partly what strong types are for .. in other words: if we check for invalid types here, the code should throw an error. i wouldn't introduce a new constant for that though but have a check within the for loop: `if(!content) throw ...`